native/src/main/java/io/ballerina/stdlib/ftp/client/FtpClient.java

```diff
@@ -90,6 +95,19 @@ public static Object initClientEndpoint(BObject clientEndpoint, BMap<Object, Obj
                     ftpConfig.put(FtpConstants.IDENTITY_PASS_PHRASE, privateKeyPassword.getValue());
                 }
             }
+            final BArray preferredMethods = auth.getArrayValue((StringUtils.fromString(
+                    ENDPOINT_CONFIG_PREFERRED_METHODS)));
+            if (preferredMethods != null) {
+                if (preferredMethods.isEmpty()) {
+                    return FtpUtil.createError(NO_AUTH_METHOD_ERROR, Error.errorType());
+                }
+                String authMethods = Arrays.stream(preferredMethods.getValues())
+                        .map(Object::toString)
+                        .map(String::toLowerCase)
```

Comment:
Can't we combine these map methods and do the conversion ? I would prefer to have a separate method which do the relevant conversion and use it inside one of the `map` operations.